tools/jsonschema_to_rust_dto_enum_flags.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python
+
+import os
+import re
+import sys
+import json
+
+from typing import Iterator, cast
+
+DEFAULT_INDENT = 4
+DOCS_URL = 'https://github.com/kamu-data/open-data-fabric/blob/master/open-data-fabric.md#{}-schema'
+ALLOWED_SCHEMAS = frozenset(['MetadataEvent'])
+
+PREAMBLE = f"""
+///////////////////////////////////////////////////////////////////////////////
+// WARNING: This file is auto-generated from Open Data Fabric Schemas
+// See: http://opendatafabric.org/
+///////////////////////////////////////////////////////////////////////////////
+
+use bitflags::bitflags;
+
+use crate::{{ {','.join(ALLOWED_SCHEMAS)} }};
+
+///////////////////////////////////////////////////////////////////////////////
+"""
+
+JsonType = dict[str, 'JsonType']
+
+
+def render(schemas_dir: str) -> None:
+    schemas = read_schemas(schemas_dir)
+
+    print(PREAMBLE)
+
+    for name in sorted(schemas.keys()):
+        if name not in ALLOWED_SCHEMAS:
+            continue
+
+        sch = schemas[name]
+
+        try:
+            if name == 'Manifest':
+                continue
+
+            print('/' * 80)
+            print(f'// {name}')
+            print('// ' + DOCS_URL.format(name.lower()))
+            print('/' * 80)
+            print()
+
+            for l in render_schema(name, sch):
+                print(l)
+            print()
+
+
+        except Exception as ex:
+            raise Exception(
+                f'Error while rendering {name} schema:\n{sch}'
+            ) from ex
+
+
+def read_schemas(schemas_dir: str) -> dict[str, JsonType]:
+    schemas = {}
+    read_schemas_rec(schemas_dir, schemas)
+    return schemas
+
+
+def read_schemas_rec(schemas_dir: str, schemas: dict[str, JsonType]) -> None:
+    for fname in os.listdir(schemas_dir):
+        path = os.path.join(schemas_dir, fname)
+
+        if os.path.isdir(path):
+            read_schemas_rec(path, schemas)
+            continue
+
+        with open(path) as f:
+            s = json.load(f)
+            fname = os.path.splitext(os.path.split(path)[-1])[0]
+            name = os.path.splitext(s['$id'].split('/')[-1])[0]
+            assert fname == name, f"{fname} != {name}"
+            schemas[name] = s
```

Comment:
I feel like none of this complexity is actually required for this script - it only needs one single schema `MetadataEvent.json` and could just load it directly.

Reply:
Simplified -- Done